lib/graphql/language/sanitized_printer.rb

```diff
@@ -60,22 +60,31 @@ def print_node(node, indent: "")
       end
 
       def print_argument(argument)
+        # We won't have type information if we're recursing into a custom scalar
+        return super if @current_input_type && @current_input_type.kind.scalar?
+
         arg_owner = @current_input_type || @current_directive || @current_field
         arg_def = arg_owner.arguments[argument.name]
 
         old_input_type = @current_input_type
         @current_input_type = arg_def.type.non_null? ? arg_def.type.of_type : arg_def.type
-        res = super
+
+        argument_value = if coerce_argument_value_to_list?(@current_input_type, argument.value)
+          [argument.value]
+        else
+          argument.value
+        end
+        res = "#{argument.name}: #{print_node(argument_value)}".dup
+
         @current_input_type = old_input_type
         res
       end
 
-      def print_list_type(list_type)
```

Comment:
This method is only used when printing list type definitions. Thus I don't think we need to track `@current_input_type` here since we'll never recurse into arguments from here. This is not strictly required for these PR fixes but it did make it harder to understand how `@current_input_type` is used.